File: Android/SMK/BirdB&S/BirdB&SAdmin/app/src/main/java/com/example/birdbsadmin/ui/transaction/DetailAdapter.java
package com.example.birdbsadmin.ui.transaction;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.birdbsadmin.R;
import com.example.birdbsadmin.koneksi.Api;
import com.example.birdbsadmin.koneksi.ApiConfig;
import com.example.birdbsadmin.model.ModelBeliDetail;
import com.example.birdbsadmin.model.ModelItem;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class DetailAdapter extends RecyclerView.Adapter<DetailAdapter.DetailViewHolder> {
    private final Context context;
    private final List<ModelBeliDetail> listItem;

    public DetailAdapter(Context context, List<ModelBeliDetail> listItem) {
        this.context = context;
        this.listItem = listItem;
    }

    @NonNull
    @Override
    public DetailViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.item_transaction, parent, false);
        return new DetailViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull DetailViewHolder holder, int position) {
        ModelBeliDetail itemPosition = listItem.get(position);

        ApiConfig service = Api.getRetrofit().create(ApiConfig.class);
        Call<List<ModelItem>> call = service.getItemId(itemPosition.getIditem());
        call.enqueue(new Callback<List<ModelItem>>() {
            @Override
            public void onResponse(@NonNull Call<List<ModelItem>> call, @NonNull Response<List<ModelItem>> response) {
                holder.tvDate.setText(response.body().get(0).getNamaitem());
                holder.tvTotal.setText("Price Per item : Rp. " + response.body().get(0).getHarga());
            }

            @Override
            public void onFailure(@NonNull Call<List<ModelItem>> call, @NonNull Throwable t) {
                Toast.makeText(context, t.getMessage(), Toast.LENGTH_SHORT).show();
            }
        });

        holder.tvName.setText("Qty : " + itemPosition.getJumlah());
        holder.tvStatus.setText("Total : Rp. " + itemPosition.getHarga());
    }

    @Override
    public int getItemCount() {
        return listItem.size();
    }

    public static class DetailViewHolder extends RecyclerView.ViewHolder {
        TextView tvName, tvDate, tvStatus, tvTotal;

        public DetailViewHolder(@NonNull View itemView) {
            super(itemView);

            tvDate = (TextView) itemView.findViewById(R.id.tvDateItem);
            tvName = (TextView) itemView.findViewById(R.id.tvNameItem);
            tvStatus = (TextView) itemView.findViewById(R.id.tvStatusItem);
            tvTotal = (TextView) itemView.findViewById(R.id.tvTotalItem);
        }
    }
}

File: Android/Aryunas/app/src/main/java/com/alcorp/aryunas/ui/order/detailOrder/DetailAdapter.kt
package com.alcorp.aryunas.ui.order.detailOrder

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.constraintlayout.widget.ConstraintLayout
import androidx.recyclerview.widget.RecyclerView
import com.alcorp.aryunas.R
import com.alcorp.aryunas.data.local.entity.OrderDetail
import com.alcorp.aryunas.util.dateToNormal
import com.alcorp.aryunas.util.timeToNormal
import com.alcorp.aryunas.util.withNumberingFormat

class DetailAdapter(val listData: List<OrderDetail>) : RecyclerView.Adapter<DetailAdapter.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_order_detail, parent, false)
        return ViewHolder(view)
    }

    @SuppressLint("SetTextI18n")
    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val orderDetail = listData[position]
        holder.tvNama.text = orderDetail.nama_kamar
        holder.tvTglIn.text = "Check In : ${dateToNormal(orderDetail.tglin)} | ${timeToNormal(orderDetail.jamin)}"
        holder.tvTglOut.text = "Check Out : ${dateToNormal(orderDetail.tglout)} | ${timeToNormal(orderDetail.jamout)}"
        holder.tvJumlahHari.text = "Jumlah Hari : ${orderDetail.total_hari}"
        holder.tvHargaPerHari.text = "Harga Per Hari : Rp. ${orderDetail.harga_kamar.toString().withNumberingFormat()}"
        holder.tvTotal.text = "Total Harga : Rp. ${orderDetail.total_harga.toString().withNumberingFormat()}"
        holder.ivDeleteSweep.visibility = View.GONE
    }

    override fun getItemCount(): Int = listData.size

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var tvNama: TextView = itemView.findViewById(R.id.tvNama)
        var tvTglIn: TextView = itemView.findViewById(R.id.tvTglIn)
        var tvTglOut: TextView = itemView.findViewById(R.id.tvTglOut)
        var tvJumlahHari: TextView = itemView.findViewById(R.id.tvHari)
        var tvHargaPerHari: TextView = itemView.findViewById(R.id.tvHargaPerHari)
        var tvTotal: TextView = itemView.findViewById(R.id.tvTotal)
        var ivDeleteSweep: ImageView = itemView.findViewById(R.id.ivDeleteSweep)
    }
}